aidial_adapter_openai/app.py

```diff
@@ -212,13 +197,48 @@ async def embedding(deployment_id: str, request: Request):
         {**creds, "api_version": api_version}
     )
 
-    return await handle_exceptions(
-        call_with_extra_body(client.embeddings.create, data)
-    )
+    return await call_with_extra_body(client.embeddings.create, data)
+
+
+@app.exception_handler(OpenAIError)
+def openai_exception_handler(request: Request, e: DialException):
+    if isinstance(e, APIStatusError):
+        r = e.response
+        return Response(
+            content=r.content,
+            status_code=r.status_code,
+            headers=r.headers,
+        )
+
+    if isinstance(e, APITimeoutError):
+        raise DialException(
+            status_code=504,
+            type="timeout",
+            message="Request timed out",
+            display_message="Request timed out. Please try again later.",
+        )
+
+    if isinstance(e, APIConnectionError):
+        raise DialException(
+            status_code=502,
+            type="connection",
+            message="Error communicating with OpenAI",
+            display_message="OpenAI server is not responsive. Please try again later.",
+        )
+
+    if isinstance(e, APIError):
+        raise DialException(
+            status_code=getattr(e, "status_code", None) or 500,
+            message=e.message,
+            type=e.type,
+            code=e.code,
+            param=e.param,
+            display_message=None,
+        )
```

Comment:
Minor:

maybe let's use match/case for that?

```python
match e:
        case APIStatusError():
            r = e.response
            return Response(
                content=r.content,
                status_code=r.status_code,
                headers=r.headers,
            )
        
        case APITimeoutError():
            raise DialException(
                status_code=504,
                type="timeout",
                message="Request timed out",
                display_message="Request timed out. Please try again later.",
            )
        
        case APIConnectionError():
            raise DialException(
                status_code=502,
                type="connection",
                message="Error communicating with OpenAI",
                display_message="OpenAI server is not responsive. Please try again later.",
            )
        
        case APIError():
            raise DialException(
                status_code=getattr(e, "status_code", None) or 500,
                message=e.message,
                type=e.type,
                code=e.code,
                param=e.param,
                display_message=None,
            )
        
        case _:
            raise DialException(
                status_code=500, 
                message="Unknown OpenAI exception",
                # =================
                # Or maybe without "openai", just "unknown_upstream_exception" 
                code="openai_unknown_exception"
            )
```

And wrap unknown exception into some `code` so we could track such errors?

Reply:
But I guess it's not necessary for this PR, since we had same logic before, without handling unknown excepions

Reply:
Yes, let's address it in another PR.

DIAL SDK handles all kinds of errors gracefully, but the adaptor isn't based on the SDK, so we have to do it manually :(